packages/auth/src/RocketChatAuth.ts

```diff
@@ -65,18 +65,27 @@ class RocketChatAuth {
     password: string;
     code?: string | number;
   }) {
-    const response = await loginWithPassword(
-      {
-        api: this.api,
-      },
-      {
-        user,
-        password,
-        code,
+    try {
```

Comment:
@abhinavkrin, I added a try-catch block in RocketChatAuth.ts because frontend error handling relies on specific error properties such as 'totp-required', 'Unauthorized', and 'totp-invalid', which are parsed from the server response. In Api.ts, within the packages/auth folder, the ApiError class wraps the server response with a message and endpoint when the status is not OK from the Rocket.Chat server response. Before parsing it into JSON, I check if it's an instance of the ApiError class because then only i can access the response property.

Handling it in EmbeddedChatApi.ts is possible, but it requires exporting the ApiError class from Api.ts in packages/auth and importing it into EmbeddedChatApi.ts in packages/api to check whether this is an instance of ApiError. The approach I took was more straightforward because we already have access to the ApiError class. Additionally, when it is not an instance of ApiError, I rethrow the error, and those cases will be handled as before.

Another reason for this approach is its similarity to the existing pattern in the getCurrentUser function in RocketChatAuth.ts, making the structure consistent rather than exporting and importing a class.

These are the reasons why I did it this way. Thank you!

Reply:
Hey @Spiral-Memory I understand, but the package auth is built keeping in mind that it could be used by other projects also. So we would want to have a uniform behavior.
There are other ways you can export the error reason.
![error](https://github.com/RocketChat/EmbeddedChat/assets/15830206/b8213abc-f487-4e55-9dcf-ab7d64d1c262)
You can pass more information though the cause. 
or
Add a code property to error object
or
Make use of the response property of ApiError

In short, the error instance should be thrown from the function instead of a simple JSON object.